extendr-api/src/wrapper/matrix.rs

```diff
@@ -118,13 +118,35 @@ where
 // @findex dimgets
 
 impl<T> RMatrix<T> {
+    pub fn get_dimnames(&self) -> Robj {
+        let dims = self.dimnames();
+
+        if dims.is_none() {
+            return ().into_robj();
+        }
+
+        dims.unwrap().into_iter().collect::<List>().into_robj()
+    }
+
     pub fn get_colnames(&self) -> Robj {
-        let colnames = Robj::from_sexp(unsafe { Rf_GetColNames(self.get()) });
-        colnames
+        let dims = self.dimnames();
+
+        if dims.is_none() {
+            return ().into_robj();
+        }
+
+        let dims = dims.unwrap().collect::<List>();
+        dims[1].clone()
     }
     pub fn get_rownames(&self) -> Robj {
-        let rownames = Robj::from_sexp(unsafe { Rf_GetRowNames(self.get()) });
-        rownames
+        let dims = self.dimnames();
+
+        if dims.is_none() {
+            return ().into_robj();
+        }
+
+        let dims = dims.unwrap().collect::<List>();
+        dims[1].clone()
```

Comment:
I don't know why you got `NULL`, but you shouldn't, see this:

```c
/* "GetRowNames" and "GetColNames" are utility routines which
 * locate and return the row names and column names from the
 * dimnames attribute of a matrix.  They are useful because
 * old versions of R used pair-based lists for dimnames
 * whereas recent versions use vector based lists.

 * These are now very old, plus
 * ``When the "dimnames" attribute is
 *   grabbed off an array it is always adjusted to be a vector.''

 They are used in bind.c and subset.c, and advertised in Rinternals.h
*/
SEXP GetRowNames(SEXP dimnames)
{
    if (TYPEOF(dimnames) == VECSXP)
    return VECTOR_ELT(dimnames, 0);
    else
    return R_NilValue;
}

SEXP GetColNames(SEXP dimnames)
{
    if (TYPEOF(dimnames) == VECSXP)
    return VECTOR_ELT(dimnames, 1);
    else
    return R_NilValue;
}
```

Reply:
Can you try your existing branch with the integration tests? The integration tests show that they weren't working for me.

Reply:
I see the issue. These `GetColNames` takes a `dimname`.. Duh.